Source/CesiumRuntime/Private/CesiumGlobeAnchorComponent.cpp

```diff
@@ -233,22 +233,38 @@ void UCesiumGlobeAnchorComponent::Sync() {
   }
 }
 
-ACesiumGeoreference* UCesiumGlobeAnchorComponent::ResolveGeoreference() {
-  if (IsValid(this->ResolvedGeoreference)) {
+ACesiumGeoreference*
+UCesiumGlobeAnchorComponent::ResolveGeoreference(bool bForceReresolve) {
+  if (IsValid(this->ResolvedGeoreference) && !bForceReresolve) {
     return this->ResolvedGeoreference;
   }
 
-  if (IsValid(this->Georeference.Get())) {
-    this->ResolvedGeoreference = this->Georeference.Get();
-  } else {
-    this->ResolvedGeoreference =
-        ACesiumGeoreference::GetDefaultGeoreference(this);
-  }
+  ACesiumGeoreference* Previous = this->ResolvedGeoreference;
+  ACesiumGeoreference* Next =
+      IsValid(this->Georeference.Get())
+          ? this->ResolvedGeoreference = this->Georeference.Get()
+          : ACesiumGeoreference::GetDefaultGeoreferenceForActor(
+                this->GetOwner());
 
-  if (this->ResolvedGeoreference) {
-    this->ResolvedGeoreference->OnGeoreferenceUpdated.AddUniqueDynamic(
-        this,
-        &UCesiumGlobeAnchorComponent::_onGeoreferenceChanged);
+  if (Previous != Next) {
+    if (IsValid(Previous)) {
+      // If we previously had a valid georeference, first synchronize using the
+      // old one so that the ECEF and Actor transforms are both up-to-date.
+      this->Sync();
```

Comment:
I like the addition of `::Sync`, here.

Because this exists, we could remove the other `::Sync` calls from line 82, and 513